linux_os/guide/system/network/network-nftables/set_nftables_base_chain/ansible/shared.yml

```diff
@@ -0,0 +1,39 @@
+# platform = multi_platform_sle
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+#Name of the table
+{{{ ansible_instantiate_variables("var_nftables_table") }}}
+#Familiy of the table 
+{{{ ansible_instantiate_variables("var_nftables_family") }}}
+#Number of base chains
+{{{ ansible_instantiate_variables("var_nftables_base_chain_number") }}}
+#Name(s) of base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_names") }}}
+#Type(s) of base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_types") }}}
+# Hooks for base chain
+{{{ ansible_instantiate_variables("var_nftables_base_chain_hooks") }}}
+#Priority
+{{{ ansible_instantiate_variables("var_nftables_base_chain_priorities") }}}
+#Policy 
+{{{ ansible_instantiate_variables("var_nftables_base_chain_policies") }}}
+
+- name: {{{ rule_title }}} - Collect Existing NFtables
+  ansible.builtin.command: nft list tables
+  register: existing_nftables
+
+- name: {{{ rule_title }}} - Set NFTable Table
+  ansible.builtin.command: nft create table {{ var_nftables_family }} {{ var_nftables_table }}
+  when: existing_nftables.stdout_lines | length == 0
```

Comment:
What happens if there is already any other table different than `filter` or an existing `filter` table is not from `inet` family? Could this happen?
Same question is valid for the bash remediation.

Reply:
Hello @marcusburghardt 

To create a chain we need to have a table in advance created, otherwise the rule will not work. There is another rule which checks/creates a table and I reused the code from this rule here. Because the rules are executed in alphabetical order - in general and because Compliance as Code does not check for dependencies - I used this approach. 

If you have any better idea/proposal, I will appreciate it.
  
Have a nice day
Rumen

Reply:
@rumch-se did you have opportunity to check this? I believe it would be important to make this conditional more robust to ensure the remediation is properly executed whenever necessary.